netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -638,11 +640,8 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
 
   @Override
   public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
-    if (gracefulShutdown == null) {
-      gracefulShutdown = new GracefulShutdown("app_requested", null);
-      gracefulShutdown.start(ctx);
-      ctx.flush();
-    }
+    gracefulClose(ctx, new GracefulServerCloseCommand("app_requested"), promise);
+    ctx.flush();
```

Comment:
This is now called unconditionally whereas the old code only called it if `gracefulShutdown == null`.

Reply:
That's fine. Flush doesn't do anything if it doesn't need to. close() needs flush because the caller won't be calling flush. The GracefulServerCloseCommand doesn't need flush because the caller of write() is responsible for calling flush (generally via `writeAndFlush(obj)`).